t/app/controller/alert_new.t

```diff
@@ -866,4 +870,66 @@ subtest 'check setting include dates in new updates cobrand option' => sub {
     $include_date_in_alert_override->restore();
 };
 
+subtest 'test notification preferences' => sub {
+    # Create a user with both email and phone verified
+    my $user1 = $mech->create_user_ok('alerts@example.com',
+        name => 'Alert User',
+        phone => '01234',
+        email_verified => 1,
+        phone_verified => 1,
+    );
+    my $user2 = $mech->create_user_ok('reporter@example.com', name => 'Reporter User');
+    my $user3 = $mech->create_user_ok('updates@example.com', name => 'Update User');
+
+    my $dt = DateTime->now->add( minutes => -30 );
+    my $r_dt = $dt->clone->add( minutes => 20 );
+
+    my ($report) = $mech->create_problems_for_body(1, $body->id, 'Testing', {
+        user => $user2,
+    });
+
+    my $update = $mech->create_comment_for_problem($report, $user3, 'Anonymous User', 'This is some more update text', 't', 'confirmed', undef, { confirmed  => $r_dt });
+
+    my $alert_user1 = FixMyStreet::DB->resultset('Alert')->create({
+        user       => $user1,
+        alert_type => 'new_updates',
+        parameter  => $report->id,
+        confirmed  => 1,
+        whensubscribed => $dt,
+    });
+    ok $alert_user1, "alert created";
+
+    # Check they get a normal email alert by default
+    FixMyStreet::Script::Alerts::send();
+    $mech->email_count_is(1);
+    is @{$twilio->texts}, 0;
+    $mech->clear_emails_ok;
+
+    # Check they don't get any update when set to none
+    FixMyStreet::DB->resultset('AlertSent')->delete;
+    $user1->update({ extra => { update_notify => 'none' } });
+    FixMyStreet::Script::Alerts::send();
+    $mech->email_count_is(0);
+    is @{$twilio->texts}, 0;
+
+    FixMyStreet::override_config {
+        ALLOWED_COBRANDS => 'fixmystreet',
+        SMS_AUTHENTICATION => 1,
+        TWILIO_ACCOUNT_SID => 'AC123',
+        MAPIT_URL => 'http://mapit.uk/',
+    }, sub {
+        FixMyStreet::DB->resultset('AlertSent')->delete;
+        $user1->update({ extra => { update_notify => 'phone' } });
+        FixMyStreet::Script::Alerts::send();
+        $mech->email_count_is(0);
+        is @{$twilio->texts}, 1;
+        my $text = $twilio->texts->[0]->{Body};
+        like $text, qr{Your report has had an update; please visit http://www.example.org/report/\d+ to view\. http://www.example.org/A/[A-Za-z0-9]+ to stop\.};
+    };
+
```

Comment:
Might be worth a test here to make sure the verified_phone and !verified_email route works too.